how-to-docs/elastic-container-registry.md

```diff
@@ -0,0 +1,143 @@
+# AWS ECR for Private Docker Images
+
+AWS ECR (Elastic Container Registry) is now available as a self-service feature via the Platform Hub. Each project has the capability to create their own Docker Repositories and define individual access to each via the use of IAM Credentials.
+
+## Creating a Docker Repository
+
+Anybody that is part of a Project within the Platform Hub will have the ability to create a new Docker Repository.
+
+1. Login to the Platform Hub via https://hub.acp.homeoffice.gov.uk
+1. Navigate to the Projects list: https://hub.acp.homeoffice.gov.uk/projects/list
+1. Select your Project from the list to go to the detail page (e.g. https://hub.acp.homeoffice.gov.uk/projects/detail/acp)
+    * Ensure you have a **Service** defined within your Project for the Docker Repository to be associated with (check under the `SERVICES` tab)
+1. Select the `ALL DOCKER REPOS` tab
+1. Select the `REQUEST NEW DOCKER REPO` button
+1. Choose the Service to associate this Repository with and provide the name of the Repository to be created (e.g. `hello-world-app`)
+
+The request to create a new Docker Repository can take a few seconds to complete. You can view the status of a Repository by navigating to the `ALL DOCKER REPOS` tab and viewing the list. Once the request has completed, your Repository should have the `Active` label associated with it.
+
+## Accessing a Docker Repository
+
+Accessing the AWS Container Registry is currently a two-step process:
+1. Use IAM Credentials to generate a temporary authorisation token
+1. Use the temporary authorisation token to authenticate your docker client with ECR, to be able to pull and push images
+
+> **Note:** The authorisation token generated for docker login is only valid for 12 hours, and so the process above will need to be repeated
+
+### Pre-Requisites
+
+To follow the below steps you must have:
+* AWS CLI (version 1.11.91 or above, check with `aws --version`)
+  * Install Guides: [Linux](https://docs.aws.amazon.com/cli/latest/userguide/install-linux.html), [OSX](https://docs.aws.amazon.com/cli/latest/userguide/install-macos.html), [Windows](https://docs.aws.amazon.com/cli/latest/userguide/install-windows.html)
+* Docker (version 17.06 or above, check with `docker --version`)
+
+### Step 1: Generate IAM Credentials
+
+1. Navigate to the `ALL DOCKER REPOS` tab for your Project within the Platform Hub
+1. For the Repository you have created, select the `MANAGE ACCESS` button
+1. At this stage, you can:
+    * Create a Robot Account, which can be used in deployment pipelines in Drone CI for publishing new images to AWS ECR
+    * Select which Project Members have the ability to pull images, and additionally push updates using their own IAM credentials (separate to the Robot Account and CI builds)
+1. In this example, select your own User, grant write access and press `Save`
```

Comment:
Think we should avoid putting this in, and state that only the robot user should  have write ideally.

Reply:
Good point, I'll add a note. I've included here just to help the guide and users testing ECR for the first time.

Reply:
I've dropped the `grant write access` and added a note, thanks 👍